src/strands/tools/registry.py

```diff
@@ -24,244 +24,318 @@
 
 
 class ToolRegistry:
-    """Central registry for all tools available to the agent.
+    """Agent Tool CRUD Interface.
 
-    This class manages tool registration, validation, discovery, and invocation.
+    Provides methods for creating, reading, updating, and deleting tools in the registry.
     """
 
-    def __init__(self) -> None:
-        """Initialize the tool registry."""
-        self.registry: Dict[str, AgentTool] = {}
-        self.dynamic_tools: Dict[str, AgentTool] = {}
-        self.tool_config: Optional[Dict[str, Any]] = None
+    def __init__(self, load_tools_from_directory: bool = False) -> None:
+        """Initialize the ToolRegistry.
 
-    def process_tools(self, tools: List[Any]) -> List[str]:
-        """Process tools list that can contain tool names, paths, imported modules, or functions.
+        Args:
+            load_tools_from_directory: Whether to load tools from the tools directory.
+                If True, tools will be loaded from the standard tool directories.
+        """
+        self.agent_tools: Dict[str, AgentTool] = {}
+        self.tool_directories = self._find_tool_directories()
+        self._tool_config: Dict[str, Any] = {"tools": []}
+        self._hot_reload_tools: Dict[str, AgentTool] = {}
+
+        if load_tools_from_directory:
+            self._load_directory_tools()
+
+    def create_tool(self, tool: Union[str, dict[str, str], Any]) -> str:
```

Comment:
I'm not sure that `create` makes sense here.  In some cases, it's wrapping an instance of an AgentTool, in other cases it's adding it directly.  IMHO, here it makes more sense that this would be called `add_tool` or simply `add`